pkg/client/group_mgmt_client.go

```diff
@@ -136,12 +167,32 @@ func (client *GroupMgmtClient) Put(path, id string, payload interface{}, respHol
 			return nil, err
 		}
 		return nil, fmt.Errorf("http response error: status (%d), messages: %v", response.StatusCode(), errResp)
+	} else {
+		// http code 202, handle async job
+		if response.StatusCode() == 202 {
+			// extract error message
+			errResp, _ := unwrapError(response.Body())
+			if client.WaitOnJob { // check sync flag
+				unwrapMessage := &ErrorResponse{}
+				err = json.Unmarshal(response.Body(), unwrapMessage)
+				if err != nil {
+					return nil, err
+				}
+				ids, err := handleAsyncJob(unwrapMessage, client)
+
+				if err != nil {
+					return nil, fmt.Errorf("http response error: status (%d), messages: %v", response.StatusCode(), err.Error())
+				}
+				return ids, fmt.Errorf("http response error: status (%d), messages: %v", response.StatusCode(), errResp)
+			}
+			return nil, fmt.Errorf("http response error: status (%d), messages: %v", response.StatusCode(), errResp)
+		}
+		return unwrap(response.Body(), respHolder)
```

Comment:
This looks like common code that we can move into a function.

Reply:
sure .